File: client/real-estate-crm-react/src/types/campaign.ts
export const CampaignChannel = {
  Email: "Email",
  WhatsApp: "WhatsApp",
} as const;
export type CampaignChannel = (typeof CampaignChannel)[keyof typeof CampaignChannel];

export const CampaignStatus = {
  Draft: "Draft",
  Sent: "Sent",
} as const;
export type CampaignStatus = (typeof CampaignStatus)[keyof typeof CampaignStatus];

export interface Campaign {
  id: string;
  name: string;
  channel: CampaignChannel;
  subject: string | null;
  body: string;
  targetStatus: string | null;
  targetSource: string | null;
  status: CampaignStatus;
  sentAt: string | null;
  recipientCount: number;
  successCount: number;
  failureCount: number;
  createdAt: string;
}

export interface CreateCampaignRequest {
  name: string;
  channel: CampaignChannel;
  subject?: string;
  body: string;
  targetStatus?: string;
  targetSource?: string;
}

export interface CampaignRecipient {
  id: string;
  leadId: string;
  success: boolean;
  errorMessage: string | null;
  sentAt: string;
}

File: src/RealEstateCRM.Domain/Entities/Campaign.cs
using RealEstateCRM.Domain.Common;
using RealEstateCRM.Domain.Enums;

namespace RealEstateCRM.Domain.Entities;

/// <summary>
/// A one-shot bulk broadcast to a segment of Leads (filtered by Status/Source), sent over
/// Email or WhatsApp. Not a scheduled/recurring drip sequence — that's a natural next step.
/// </summary>
public class Campaign : TenantEntity
{
    public string Name { get; set; } = string.Empty;
    public CampaignChannel Channel { get; set; }
    public string? Subject { get; set; }
    public string Body { get; set; } = string.Empty;
    public LeadStatus? TargetStatus { get; set; }
    public LeadSource? TargetSource { get; set; }
    public CampaignStatus Status { get; set; } = CampaignStatus.Draft;
    public Guid CreatedByUserId { get; set; }
    public DateTime? SentAt { get; set; }
    public int RecipientCount { get; set; }
    public int SuccessCount { get; set; }
    public int FailureCount { get; set; }
}
